peerpodconfig-ctrl/config/rbac/role.yaml

```diff
@@ -55,6 +55,27 @@ rules:
   - patch
   - update
   - watch
+- apiGroups:
+  - apps
+  resources:
+  - daemonsets
+  - deployments
+  - replicasets
+  - statefulsets
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - apps
+  resources:
```

Comment:
Same as above, shall we use `resourceNames` to limit the resources instances here?

Reply:
Like setting it to the following - https://github.com/confidential-containers/cloud-api-adaptor/blob/main/peerpodconfig-ctrl/controllers/peerpodconfig_controller.go#L139 ?

Reply:
@bpradipt Like https://github.com/confidential-containers/cloud-api-adaptor/blob/main/peerpod-ctrl/config/rbac/role.yaml#L11
As explained as below in https://kubernetes.io/docs/reference/access-authn-authz/rbac/
```
You can also refer to resources by name for certain requests through the resourceNames list. When specified, requests can be restricted to individual instances of a resource.
```

Reply:
I guess peerpodconfig-ctrl should only be grant necessary permission to change specific instances which should be limited by `resourceNames` list?

Reply:
I updated the PR to explicitly mention the daemonset created